prov/efa/src/rdm/rxr_ep.c

```diff
@@ -2197,6 +2197,17 @@ bool rxr_ep_use_shm_for_tx(struct fi_info *info)
 	    && !(info->caps & FI_LOCAL_COMM))
 		return 0;
 
+	/*
+	 * shm provider must make cuda calls to transfer cuda memory.
+	 * if cuda call is not allowed, we cannot use shm for transfer.
+	 * Note that, the other 2 hmem interfaces supported by EFA:
+	 * AWS Neuron and Habans synapse, will not pass intra-node traffic
+	 * to EFA (and shm provider does not support them either),
```

Comment:
I think this comment is misleading.

I think the change is good for CUDA, but perhaps we should limit it to only when CUDA is selected, and leave a comment about maybe disabling it for other HMEM types as well.

Reply:
> I think the change is good for CUDA, but perhaps we should limit it to only when CUDA is selected

In the info object, there is no information about which type of interface the application is going to use.